zerver/lib/test_fixtures.py

```diff
@@ -386,3 +386,49 @@ def remove_test_run_directories(expiry_time: int = 60 * 60) -> int:
             except FileNotFoundError:
                 pass
     return removed
+
+
+def reset_zulip_test_database() -> None:
+    """
+    This function is used to reset the zulip_test database fastest way possible,
+    i.e. First, it deletes the database and then clones it from zulip_test_template.
+    This function is used with puppeteer tests, so it can quickly reset the test
+    database after each run.
+    """
+    from zerver.lib.test_runner import destroy_test_databases
+
+    # Make sure default database is 'zulip_test'.
+    assert connections["default"].settings_dict["NAME"] == "zulip_test"
+
+    # Clearing all the active PSQL sessions with 'zulip_test'.
+    run(
+        [
+            "env",
+            "PGHOST=localhost",
+            "PGUSER=zulip_test",
+            "scripts/setup/terminate-psql-sessions",
+            "zulip_test",
+        ]
+    )
+
+    destroy_test_databases()
+    # Pointing default database to test database template, so we can instantly clone it.
+    settings.DATABASES["default"]["NAME"] = settings.BACKEND_DATABASE_TEMPLATE
+    connection = connections["default"]
+    clone_database_suffix = "clone"
+    connection.creation.clone_test_db(
+        suffix=clone_database_suffix,
+    )
```

Comment:
It's great that you got something working here!  The code is definitely messy, but that's OK for a first version of this sort of tooling.

If we can do it, what I'd like to see as an iteration on this would be to use the `worker_id` mechanism to have the test databases that we're using / messing with here not be the main `zulip_test` database name.  I'm not sure how easy that will be (`test_runner.py` has a bunch of really careful logic to avoid leaking copies of the database that @whoodes helped build, so CCing him in case he has thoughts).

Reply:
I'd like to work on it. I think so we can re-use some of the functions of `test_runner.py`
I have some doubt about this approach: 
- @timabbott  by `worker_id` mechanism, you mean we will use `worker_id` for individual test or puppeteer test process (when we want to run multiple puppeteer test parallelly.)
- How will we point our running test server to another DB (`zulip_test_database_id`).

I also like to know what @whoodes think is the best path forward.